src/sidebar/components/test/help-panel-test.js

```diff
@@ -1,45 +1,197 @@
+    it('should show navigation link back to tutorial sub-panel', () => {
+      const wrapper = createComponent();
+      wrapper.find('.help-panel__sub-panel-link').simulate('click');
+
+      const link = wrapper.find('.help-panel__sub-panel-link');
 
-    angular.module('h', []).component('helpPanel', require('../help-panel'));
+      assert.isTrue(wrapper.find('VersionInfo').exists());
+      assert.isFalse(wrapper.find('Tutorial').exists());
+      assert.equal(link.text(), 'Getting started');
+    });
+
+    it('should switch to tutorial sub-panel when link clicked', () => {
+      const wrapper = createComponent();
+
+      // Click to get to version-info sub-panel...
+      wrapper.find('.help-panel__sub-panel-link').simulate('click');
+
+      const link = wrapper.find('.help-panel__sub-panel-link');
+      // Click again to get back to tutorial sub-panel
+      link.simulate('click');
+
+      assert.isFalse(wrapper.find('VersionInfo').exists());
+      assert.isTrue(wrapper.find('Tutorial').exists());
+    });
+  });
+
+  describe('`HelpPanelTab`s', () => {
+    it('should render static link to knowledge base', () => {
+      const wrapper = createComponent();
 
-    angular.mock.module('h', {
-      store: fakeStore,
-      serviceUrl: sinon.stub(),
+      assert.isTrue(
+        wrapper
+          .find('HelpPanelTab')
+          .filter({ linkText: 'Help topics' })
+          .exists()
+      );
     });
 
-    angular.mock.inject(function(_$componentController_, _$rootScope_) {
-      $componentController = _$componentController_;
-      $rootScope = _$rootScope_;
+    it('should render dynamic link to create a new help ticket', () => {
+      const wrapper = createComponent();
+      const helpTab = wrapper
+        .find('HelpPanelTab')
+        .filter({ linkText: 'New support ticket' });
+      assert.isTrue(helpTab.exists());
+      assert.include(helpTab.prop('url'), 'fakeURLString');
     });
   });
 
-  it('displays the URL and fingerprint of the first connected frame', function() {
-    fakeStore.frames.returns([
-      {
-        uri: 'https://publisher.org/article.pdf',
-        metadata: {
-          documentFingerprint: '12345',
-        },
-      },
-    ]);
-
-    const $scope = $rootScope.$new();
-    const ctrl = $componentController('helpPanel', { $scope: $scope });
-    $scope.$digest();
-
-    assert.equal(ctrl.url, 'https://publisher.org/article.pdf');
-    assert.equal(ctrl.documentFingerprint, '12345');
+  context('dismissing the tutorial and clearing profile setting', () => {
+    context('profile preference to auto-show tutorial is truthy', () => {
+      beforeEach(() => {
+        fakeStore.getState.returns({
+          session: { preferences: { show_sidebar_tutorial: true } },
+        });
+      });
+
+      it('should not dismiss the panel when it is initially opened', () => {
+        const wrapper = createComponent();
+        const onActiveChanged = wrapper
```

Comment:
Figuring out how to test this was _challenging_. I'm still not clear on the mount type of the `SidebarPanel` wrapping element...but...eh, it works.

However, I have a sense that making sense of these tests could be a little...also challenging. Let me know if there are helpful comments or organization that could make these clearer.

Reply:
No hits for `wrapper.update` in this test module—did you mean a different one?

Reply:
oops! my bad